File: solutions/c/space-age/1/space_age.c
#include "space_age.h"

#define SECONDS_PER_EARTH_YEAR (60 * 60 * 24 * 365.25)

const float ORBITAL_PERIOD[] = {
    0.2408467,  // Mercury
    0.61519726, // Venus
    1.0,        // Earth
    1.8808158,  // Mars
    11.862615,  // Jupiter
    29.447498,  // Saturn
    84.016846,  // Uranus
    164.79132,  // Neptune
};

float age(planet_t planet, int64_t seconds)
{
    if (planet < MERCURY || planet > NEPTUNE)
        return ERROR;

    return seconds / SECONDS_PER_EARTH_YEAR / ORBITAL_PERIOD[planet];
}

File: solutions/python/space-age/3/space_age.py
from functools import partial

ORBITAL_PERIOD = {
    "mercury": 0.2408467,
    "venus": 0.61519726,
    "earth": 1.0,
    "mars": 1.8808158,
    "jupiter": 11.862615,
    "saturn": 29.447498,
    "uranus": 84.016846,
    "neptune": 164.79132,
}

SECONDS_PER_EARTH_YEAR = 31_557_600


class SpaceAge:
    _years: float

    def __init__(self, seconds: int) -> None:
        self._years = seconds / SECONDS_PER_EARTH_YEAR
        for planet, ratio in ORBITAL_PERIOD.items():
            setattr(self, "on_" + planet, partial(self._on_planet, ratio))

    def _on_planet(self, ratio: float) -> float:
        return round(self._years / ratio, 2)
